internal/cache/metadata/stat_cache.go

```diff
@@ -79,13 +80,29 @@ type statCacheBucketView struct {
 type entry struct {
 	o          *gcs.Object
 	expiration time.Time
+	key        string
 }
 
-// Size returns the size of entry.
-// It is currently set to dummy value 1 to avoid
-// the unnecessary actual size calculation.
-func (e entry) Size() uint64 {
-	return 1
+// Size returns the memory-size (resident set size) of the receiver entry.
+// The size calculated by the unsafe.Sizeof calls, and
+// NestedSizeOfGcsObject etc. does not account for
+// hidden memembers in data structures like maps, slices, linked-lists etc.
+// To account for those, we are adding a fixed constant of 553 bytes (deduced from
+// benchmark runs) to heap-size per positive stat-cache entry
+// to calculate a size closer to the actual memory utilization.
+func (e entry) Size() (size uint64) {
+	// First, calculate size on heap.
+	size = uint64(util.UnsafeSizeOf(&e) + len(e.key) + 2*util.UnsafeSizeOf(&e.key) + util.NestedSizeOfGcsObject(e.o))
+	if e.o != nil {
+		size += 553
+	}
+	// Additional 2*util.UnsafeSizeOf(&e.key) is to account for the copies of string
```

Comment:
this comment should be above the calculation part.

Reply:
done